src/rust/engine/src/tasks.rs

```diff
@@ -16,14 +17,12 @@ pub struct Task {
 }
 
 /**
- * Registry of tasks able to produce each type, along with a few fundamental python
- * types that the engine must be aware of.
+ * Registry of Tasks able to produce each type, and Singletons which are the default/only
```

Comment:
I think this should just say 'only provider' since singletons dispatch before tasks.